### PR TITLE
Add `benchmark::DoNotOptimize(...)` to help users prevent optimizations.

### PR DESCRIPTION
**Pull request by EricWF**
Calling `DoNotOptimize(Tp& value)` should be sufficient to prevent the compiler from optimizing away value and the expression that created it.

### CODE REVIEW

File: test/basic_test.cc [hunk 1]
```diff
@@ -8,8 +8,7 @@
```
Comment (dominichamon) [inline test/basic_test.cc]: i think you can use this in the other benchmarks too with the CalculatePi benchmark.

File: test/benchmark_test.cc [hunk 2]
```diff
@@ -84,9 +84,8 @@ BENCHMARK_RANGE(BM_CalculatePiRange, 1, 1024 * 1024);
 
 static void BM_CalculatePi(benchmark::State& state) {
   static const int depth = 1024;
-  double pi BENCHMARK_UNUSED = 0.0;
```
Comment (dominichamon) [inline test/benchmark_test.cc]: do we need BENCHMARK_UNUSED to be public any more?

Comment (EricWF) [inline test/benchmark_test.cc]: I think so. BENCHMARK_UNUSED is used in the `BENCHMARK()` macros.

File: test/basic_test.cc (hunk 1 above)
Comment (EricWF) [inline test/basic_test.cc]: ack.

**Comment by EricWF** [comment]
I buggered the revision history on this branch. Commited via patch.

**Pull request closed without merging**